src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineHelper.cs

```diff
@@ -349,13 +314,33 @@ public static void ExpandAncestors(ImmutableArray<DocumentSymbolUIItem> document
             }
         }
 
-        internal static void UnselectAll(IEnumerable<DocumentSymbolUIItem> documentSymbolItems)
+        internal static void UnselectAll(ImmutableArray<DocumentSymbolDataViewModel> documentSymbolItems)
         {
             foreach (var documentSymbolItem in documentSymbolItems)
             {
                 documentSymbolItem.IsSelected = false;
                 UnselectAll(documentSymbolItem.Children);
             }
         }
+
+        internal static bool AreAllTopLevelItemsCollapsed(ImmutableArray<DocumentSymbolDataViewModel> documentSymbolViewModelItems)
+        {
+            if (!documentSymbolViewModelItems.Any())
+            {
+                // We are operating on an empty array this can happen if the LSP service hasn't populated us with any data yet.
+                return false;
```

Comment:
still feels weird.  note to self: have to examien callers of this to see what they do with false.

Reply:
this is still unclear.  method has 'All' naming but doesn't follow .net logic for `.All`.  What happens if true is returned from this?